### DISCUSSION TITLE
Nextjs 14 - Getting the referer as null when reading from the next headers in Middleware

### DISCUSSION THREAD

**Discussion by ygg-jay** [vercel/next.js#71969] [Help]
### Summary

In nextjs 14 I'm trying to read referer header in middleware but it's value is null. But at times I'm getting the value. Any idea what might be wrong.

### Additional information

```js
const headersList = headers();
const headerRef = headersList.get('referer');
```

**Comment by icyJoseph** [comment]
Hi,

Could it be that some pages have a Referrer Policy set? https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referrer-Policy

**Comment by aniruddhaadak80** [comment]
Hi @ygag-jay ,

The referer header sometimes returns null in Next.js 14 middleware likely due to:

1. Browser Privacy Settings: Browsers may omit referer based on referrer policies, especially with cross-origin requests.
2. Caching Issues: CDN or edge caching might strip certain headers.
3. Referrer Policy: Check if the page or request has a restrictive referrer policy (e.g., no-referrer).
4. Redirects: Server-side redirects can sometimes drop headers.


**Example Troubleshooting Code :**

Add logging to check the referer availability:
```
export function middleware(request) {
  const referer = request.headers.get('referer');
  
  console.log('Referer:', referer); // Logs 'Referer: null' if missing
  
  return NextResponse.next();
}
```

Try logging referer across different environments and inspect referrer policies in requests.
